src/metabase/models/permissions.clj

```diff
@@ -826,27 +842,57 @@
+  (->>
+   graph
+   (walk/postwalk-replace {{:query {:schemas :all}}             {:query {:schemas :all :native :none}}
+                           {:query {:schemas :all :native nil}} {:query {:schemas :all :native :none}}})))
+
+(mu/defn generate-graph :- :map
+  "Used to generation permission graph from parsed permission paths of v1 and v2 permission graphs for the api layer."
+  [db-ids group-id->paths :- [:map-of :int [:* Path]]]
+  (->> group-id->paths
+       (m/map-vals
+        (fn [paths]
+          (let [permissions-graph (perms-parse/->graph paths)]
+            (if (= permissions-graph :all)
+              (all-permissions db-ids)
+              (:db permissions-graph)))))
+       post-process-graph))
+
 (defn data-perms-graph
   "Fetch a graph representing the current *data* permissions status for every Group and all permissioned databases.
   See [[metabase.models.collection.graph]] for the Collection permissions graph code."
   []
-  (let [group-id->paths (permissions-by-group-ids [:or
-                                                   [:= :object (h2x/literal "/")]
-                                                   [:like :object (h2x/literal "%/db/%")]])
-        db-ids          (delay (db/select-ids 'Database))
-        group-id->graph (m/map-vals
-                         (fn [paths]
-                           ;; Currently we do not use v2 permissions paths, and permissions->graph doesn't handle them.
-                           ;; so we ignore those until that work is complete.
-                           (let [v1-paths (filter #(mc/validate path-regex-v1 %) paths)
-                                 permissions-graph (perms-parse/permissions->graph v1-paths)]
-                             (if (= permissions-graph :all)
-                               (all-permissions @db-ids)
-                               (:db permissions-graph))))
-                         group-id->paths)]
+  (let [db-ids             (delay (db/select-ids 'Database))
+        group-id->v1-paths (->> (permissions-by-group-ids [:or
+                                                           [:= :object (h2x/literal "/")]
+                                                           [:like :object (h2x/literal "%/db/%")]])
+                                ;;  keep v1 paths, implicitly remove v2
+                                (m/map-vals (fn [paths]
+                                              (filter (fn [path]
+                                                        (mc/validate [:re path-regex-v1] path))
+                                                      paths))))]
     {:revision (perms-revision/latest-id)
-     :groups   group-id->graph}))
+     :groups   (generate-graph @db-ids group-id->v1-paths)}))
+
+(defn data-perms-graph-v2
+  "Fetch a graph representing the current *data* permissions status for every Group and all permissioned databases.
+  See [[metabase.models.collection.graph]] for the Collection permissions graph code."
```

Comment:
Probably worth clarifying the difference between this and `data-perms-graph-v1` here

Reply:
Good idea, I'll add that in.